sources/lib/reasoners/intervalCalculus.ml

```diff
@@ -1655,17 +1711,52 @@ let default_case_split env uf ~for_model =
     end
   | res -> res
 
+(** Add relation between term x and the terms in it. This can allow use to track
+    if x is computable when his subterms values are known.
+    This is currently only done for power *)
+let add_used_by t env =
+  match E.term_view t with
+  | E.Not_a_term _ -> assert false
+  | E.Term
+      { E.f = (Sy.Op Sy.Pow); xs = [a; b]; ty; _ } ->
+    begin
+      match calc_pow a b ty env.new_uf with
+      | Some (res,ex) ->
+        begin
+          match X.term_extract res with
```

Comment:
No, `X.term_extract res` may fail here.
`add_used_by` should also take as argument `rt` (make of `t`). Then, we would return an equality `rt = res` on semantic values as before.